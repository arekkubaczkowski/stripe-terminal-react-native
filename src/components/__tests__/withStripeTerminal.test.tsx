import * as React from 'react';
import { withStripeTerminal } from '../withStripeTerminal';
import { render } from '@testing-library/react-native';
import { Text, TouchableOpacity } from 'react-native';

jest.mock(
  '../../../node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter'
);

describe('withStripeTerminal.tsx', () => {
  it('renders correctly', () => {
    const MockComponent = ({}) => {
      return (
        <TouchableOpacity>
          <Text>test</Text>
        </TouchableOpacity>
      );
    };

    const WithStripeTerminalComponent = withStripeTerminal(MockComponent);

    const { toJSON, findByText } = render(<WithStripeTerminalComponent />);

    const childText = findByText('test');
    expect(childText).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
