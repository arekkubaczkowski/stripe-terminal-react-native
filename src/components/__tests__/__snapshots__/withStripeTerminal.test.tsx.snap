// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withStripeTerminal.tsx renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    test
  </Text>
</View>
`;
